hive_flutter/lib/src/hive_builder.dart

```diff
@@ -0,0 +1,57 @@
+part of hive_flutter;
+
+/// Signature for a function that builds a widget with an error [Object].
+typedef ErrorBoxWidgetBuilder = Widget Function(
+    BuildContext context, Object error);
+
+/// Signature for a function that builds a widget when the [Box] is
+/// currently loading.
+typedef LoadingBoxWidgetBuilder = Widget Function(BuildContext context);
+
+/// A general-purpose widget which rebuilds itself when the [Box] in the
+/// [openBox] function is loaded. While the box is not loaded a placeholder
+/// [placeholderBuilder] is specified to inform the user. In the case of the
+/// box failing to open an error widget [errorBuilder] is displayed.
+class HiveBuilder extends StatelessWidget {
```

Comment:
The `HiveBuilder` was initially part of `hive_flutter` because I found it useful. But it is not much more than a fancy `FutureBuilder`. Since handling `Futures` is a general Flutter problem and has nothing to do with Hive, I'm not sure if it should be included.